tests/Doctrine/Tests/DBAL/Functional/Platform/DateExpressionTest.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Doctrine\Tests\DBAL\Functional\Platform;
+
+use DateTimeImmutable;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\Tests\DbalFunctionalTestCase;
+use function sprintf;
+
+class DateExpressionTest extends DbalFunctionalTestCase
+{
+    /**
+     * @test
+     * @dataProvider dateDiffProvider
+     */
+    public function diff(string $date1, string $date2, int $expected)
+    {
+        $table = new Table('date_expr_test');
+        $table->addColumn('date1', 'date');
+        $table->addColumn('date2', 'date');
+        $this->_conn->getSchemaManager()->dropAndCreateTable($table);
+        $this->_conn->insert('date_expr_test', [
+            'date1' => $date1,
+            'date2' => $date2,
+        ]);
+
+        $platform = $this->_conn->getDatabasePlatform();
+
+        $sql  = sprintf('SELECT %s FROM date_expr_test', $platform->getDateDiffExpression('date1', 'date2'));
+        $diff = $this->_conn->query($sql)->fetchColumn();
+
+        self::assertEquals($expected, $diff);
+    }
+
+    public static function dateDiffProvider()
+    {
+        $date1    = new DateTimeImmutable();
+        $date2    = new DateTimeImmutable('2018-04-10 10:10:10');
+        $expected = $date1->modify('midnight')->diff(
+            $date2->modify('midnight')
+        )->days;
+
+        return [
```

Comment:
Could use `yield`s too. :stuck_out_tongue_winking_eye:

Reply:
i quite like it, it saves one nesting level and nested arrays of arrays (of arrays). But functionally it's irrelevant.